tests/test_modeling_tapas.py

```diff
@@ -0,0 +1,391 @@
+# coding=utf-8
+# Copyright 2018 XXX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+
+import unittest
+
+import random
+
+from transformers import is_torch_available
+from transformers.testing_utils import require_torch, slow, torch_device
+
+from .test_configuration_common import ConfigTester
+from .test_modeling_common import ModelTesterMixin, floats_tensor, ids_tensor, random_attention_mask
+
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (
+        TAPAS_PRETRAINED_MODEL_ARCHIVE_LIST,
+        TapasConfig,
+        TapasModel,
+        TapasForMaskedLM,
+        TapasForSequenceClassification,
+        TapasForQuestionAnswering,
+    )
+
+global_rng = random.Random()
+
+
+def token_type_ids_tensor_tapas(shape, type_vocab_sizes, rng=None, name=None):
+    #  Creates a random int32 tensor of the shape within the type vocab sizes
+    if rng is None:
+        rng = global_rng
+
+    values = []
+    for example in range(shape[0]):
+        for token in range(shape[1]):
+            for idx in range(shape[2]):
+                values.append(rng.randint(0, type_vocab_sizes[idx] - 1))
+
+    return torch.tensor(data=values, dtype=torch.long, device=torch_device).view(shape).contiguous()
```

Comment:
Shouldn't need to use this. You can use the `ids_tensor` you have imported previously to generate random values of a given shape, bound to certain values.

Reply:
Fixed, now using `ids_tensor` function.